code/lib/cli/src/js-package-manager/JsPackageManager.ts

```diff
@@ -264,15 +264,15 @@ export abstract class JsPackageManager {
         }
       });
 
-      this.writePackageJson(packageJson);
-    } else {
-      try {
-        this.runRemoveDeps(dependencies);
-      } catch (e) {
-        logger.error('An error occurred while removing dependencies.');
-        logger.log(e.message);
-        throw new HandledError(e);
-      }
+      return this.writePackageJson(packageJson);
+    }
+
+    try {
+      return this.runRemoveDeps(dependencies);
```

Comment:
Since this statement is not awaited, the surrounding catch block doesn’t have any effect.

```javascript
function getPromise() {
  // This promise rejects immediately
  return new Promise((resolve, reject) => reject(new Error("An error occurred")));
}

try {
  return getPromise(); // This won't be caught by the catch block below
} catch (err) {
  console.log("Caught an error:", err);
}
```

Reply:
Great catch!